client/src/js/modules/runs/views/overview.js

```diff
@@ -0,0 +1,53 @@
+define([
+    'marionette',
+
+    'collections/bls',
+    
+    'modules/stats/views/beamline',
+    
+    'views/filter',
+    'templates/runs/overview.html',
+    'jquery'
+], function(Marionette, Beamlines, BeamlineStatsView, FilterView, template, $) {
+    return Marionette.LayoutView.extend({
+        template: template,
+        className: 'content',
+        regions: {
+            pills: '.beamline-pills',
+            stats: '.beamline-stats'
+        },
+        showBeamlineStats: function(beamline) {
+            var regexExp = new RegExp(/\/bl\/.*/)
+            var url = window.location.pathname
+            if (beamline) {
```

Comment:
Please could you provide some comments about why this function needs to use the window push state function.